### ISSUE TITLE
Dinamicaly adding/removing MenuItem-s in Menu leads to memory leaks

### ISSUE THREAD

**Issue by ds1709** [AvaloniaUI/Avalonia#15990]
### Describe the bug

When adding `MenuItem`-s into `Menu` and then removing them, `Menu` keep holding reference to removed `MenuItem`-s.
Here screenshots from memory profiler.
![image](https://github.com/AvaloniaUI/Avalonia/assets/5591663/39df0433-ccdf-448b-8e16-d45122c19bbe)
and quick watch of MenuItem (Parent is null)
![image](https://github.com/AvaloniaUI/Avalonia/assets/5591663/1fd705e6-ce79-4bdc-a25d-dc851ee310bc)
and also quick watch of Menu (Items collection is empty)
![image](https://github.com/AvaloniaUI/Avalonia/assets/5591663/5b5d54a1-ef8a-4928-b896-5c4ba9098778)
As you can see, `Menu` keep holding reference on removed `MenuItem` throught chain of strong event subscriptions.

### To Reproduce

```csharp
            var w = new Window1(); // Custom window with Menu.

            var mi = new MenuItem { Header = "FooBar" };
            mi.Click += (_, _) => w.MainMenu.Items.Remove(mi);

            w.MainMenu.Items.Add(mi);
```

### Avalonia version

11.0.10

**Comment by timunie** [comment]
@ds1709 probably worth to file a PR

**Comment by ds1709** [comment]
@timunie I need this fix in version 11.0.*. So, do I need to create two PR, one in branch 11.0 and another in develop?

**Comment by ds1709** [comment]
@timunie is there any problem with support 11.0 version? This fix is not complex, so release 11.0.11 should be easy.

**Comment by timunie** [comment]
Limited resources. Nothing else.

**Comment by ds1709** [comment]
Is there any chance to get this fix in version 11.0? And another question, when is it planned to release version 11.1 (no beta)? According to the commot graph, there are a big amount of improvements from release 11.1.0-beta2. Same for 11.0.10, there are improvements, that wasn't released.

**Comment by timunie** [comment]
1. You can request 11.0 backports if you have paid support. Reach out via website if you are interested in general 
2. 11.0.10 is actually relesed. You may need to pull templates again 
3. 11.1 should drop soonish if there's no regression found while rc testings. No actual due date

Thanks for your understanding
